libs/LedKit/source/Sleeping.cpp

```diff
@@ -0,0 +1,210 @@
+// Leka - LekaOS
+// Copyright 2022 APF France handicap
+// SPDX-License-Identifier: Apache-2.0
+
+// +     Stage 1   +       Stage 2       +   3   + 4  +  5  + 6 +  7   + 8 +   9    +
+// |               |                     |       |    |     |   |      |   |        |
+// |               |                     |       |    |     |   |      |   |        |
+// |               |---------------------|       |    |     |   |      |   |        |  -------White
+// |               |                     |-\     |    |     |   |      |   |        |
+// |             -/|                     |  \    |    |     |   |      |   |        |
+// |            /  |                     |   \   |    |     |   |      |   |        |
+// |           /   |                     |    -\ |    |\    |   |      |   |        |
+// |          /    |                     |      \|  -/| \   |   |      |   |        |
+// |         /     |                     |       |-/  |  -\ |   |\     |   |        |
+// |       -/      |                     |       |    |    \| -/| -\   |   |        |
+// |      /        |                     |       |    |     |/  |   \  |   |\       |
+// |     /         |                     |       |    |     |   |    -\| -/| -\     |
+// |    /          |                     |       |    |     |   |      |/  |   -\   |
+// |  -/           |                     |       |    |     |   |      |   |     \  |
+// | /             |                     |       |    |     |   |      |   |      -\|
+// |/              |                     |       |    |     |   |      |   |        |
+// |               |                     |       |    |     |   |      |   |        |   --------Black
+
+// ? LCOV_EXCL_START - Exclude from coverage report
+
+#include "Sleeping.h"
+
+#include "LogKit.h"
+#include "MathUtils.h"
+
+namespace leka::animation {
+
+static constexpr auto in_max = 100;
+
+Sleeping::Sleeping(interface::LED &ears, interface::LED &belt) : _ears(ears), _belt(belt) {}
+
+void Sleeping::start()
+{
+	_ears.setColor(RGB::black);
+	_belt.setColor(RGB::black);
+	_ears.show();
+	_belt.show();
+}
+
+void Sleeping::stop()
+{
+	_ears.setColor(RGB::black);
+	_belt.setColor(RGB::black);
+	_ears.show();
+	_belt.show();
+}
+
+void Sleeping::run()
+{
+	stage1();
+	stage2();
+	stage3();
+	stage4();
+	stage5();
+	stage6();
+	stage7();
+	stage8();
+	stage9();
+	_belt.show();
+}
+
+auto Sleeping::mapStep(uint8_t in_max) const -> float
+{
+	return utils::math::map(_step, static_cast<uint8_t>(0), in_max, 0.F, 1.F);
+}
```

Comment:
elle fait quoi cette fonction? elle sert à quoi?

```suggestion
auto Sleeping::mapStep(uint8_t val) const -> float
{
	return utils::math::map(_step, static_cast<uint8_t>(0), val, 0.F, 1.F);
}
```

Reply:
je vois que tu passes in_max qui est une constante alors que tu utilises _step qui change à chaque fois, c'est un peu un code smell.

```suggestion
auto Sleeping::mapStep(uint8_t step) const -> float
{
	const auto kMax = uint8_t {100};
	return utils::math::map(step, uint8_t {0}, kMax, 0.F, 1.F);
}
```

que tu appelles après en passant `_step`